utils/src/main/java/com/cloud/utils/UriUtils.java

```diff
@@ -391,4 +391,57 @@ public static InputStream getInputStreamFromUrl(String url, String user, String
             return null;
         }
     }
+
+    /**
+     * Expands a given vlan URI to a list of vlan IDs
+     * @param vlanAuthority the URI part without the vlan:// scheme
+     * @return returns list of vlan integer ids
+     */
+    public static List<Integer> expandVlanUri(final String vlanAuthority) {
+        final List<Integer> expandedVlans = new ArrayList<>();
+        if (vlanAuthority == null || vlanAuthority.isEmpty()) {
+            return expandedVlans;
+        }
+        for (final String vlanPart: vlanAuthority.split(",")) {
+            if (vlanPart == null || vlanPart.isEmpty()) {
+                continue;
+            }
+            final String[] range = vlanPart.split("-");
+            if (range.length == 2) {
+                Integer start = NumbersUtil.parseInt(range[0], -1);
+                Integer end = NumbersUtil.parseInt(range[1], -1);
+                if (start <= end && end > -1 && start > -1) {
+                    while (start <= end) {
+                        expandedVlans.add(start++);
+                    }
+                }
+            } else {
+                final Integer value = NumbersUtil.parseInt(range[0], -1);
+                if (value > -1) {
+                    expandedVlans.add(value);
+                }
+            }
+        }
+        return expandedVlans;
+    }
+
+    /**
+     * Checks if given vlan URI authorities overlap
+     * @param vlanRange1
+     * @param vlanRange2
+     * @return true if they overlap
+     */
+    public static boolean checkVlanUriOverlap(final String vlanRange1, final String vlanRange2) {
+        final List<Integer> vlans1 = expandVlanUri(vlanRange1);
+        final List<Integer> vlans2 = expandVlanUri(vlanRange2);
+        if (vlans1 == null || vlans2 == null) {
```

Comment:
if one of them is null, the result is `true`? This means that they overlap?

Reply:
We're checking if two vlan ranges (comma separated ranges or values, such as `100-200,300` or `20-30` etc) overlap, if any of them when expanded (i.e. `1-3` expands to `1,2,3`) is null, i.e. there is no overlap.

Reply:
I did not understand the explanations :(

Anyways, looking at the code of `expandVlanUri`, I did not see a way for it to return null. Worst case scenario it returns an empty list. Do we need this check?

Reply:
Yes, we need this check.

Reply:
Do you mind explaining why if the method `expandVlanUri` returns an empty list in the worst case?

Reply:
In general, this case won't be hit but the implementation was done to be defensive of all sorts of inputs, including nulls. If vlanRange provided to `expandVlanUri` had issues such as non-integers or negative values, then you'll get empty list. Again, this was defensive programming.

Reply:
I understand that, but I think we should not over code or engineer. If we worry about the method `expandVlanUri` returning null, we can do something else to catch this.

What about a test case for `expandVlanUri ` that fails if it returns null? Then, we can remove this check. On thing is to be defensive when a `null` case can happen, the other is to code expecting someone to make a mistake in the future (for that it is better to write unit test cases).

Reply:
I think we're over-discussing, not over-engineering :) I'll ping you on respective unit tests.

Reply:
Ah that as well ;)

I am sorry to bother, but if I do not understand something I keep asking until I can move along.

Reply:
You could also use !Collections.disjoint(vlans1, vlans2)

Reply:
@fmaximus fixed.